validator/server_api/json.go

```diff
@@ -146,20 +145,20 @@ func ValidationInputFromJson(entry *InputJSON) (*validator.ValidationInput, erro
 		}
 		valInput.BatchInfo = append(valInput.BatchInfo, decInfo)
 	}
-	for moduleHash, info := range entry.UserWasms {
-		asm, err := base64.StdEncoding.DecodeString(info.Asm)
-		if err != nil {
-			return nil, err
-		}
-		module, err := base64.StdEncoding.DecodeString(info.Module)
-		if err != nil {
-			return nil, err
-		}
-		decInfo := state.ActivatedWasm{
-			Asm:    asm,
-			Module: module,
+	for arch, wasms := range entry.UserWasms {
+		archWasms := make(map[common.Hash][]byte)
+		for moduleHash, encoded := range wasms {
+			decoded, err := base64.StdEncoding.DecodeString(encoded)
+			if err != nil {
+				return nil, err
+			}
+			uncompressed, err := arbcompress.Decompress(decoded, 30000000)
```

Comment:
do we have a guarantee that the wasm won't exceed the `30000000` bytes? currently the uncompressed data size is not checked before compression in `ValidationInputToJson` -- should we also check the limit there?

+ nit: we might want to use some named constant for the max size `30000000` (?)